Python/Product/TestAdapter.Executor/TestExecutor.cs

```diff
@@ -94,31 +100,24 @@ class TestExecutor : ITestExecutor {
 
             var sourceToProjSettings = RunSettingsUtil.GetSourceToProjSettings(runContext.RunSettings);
 
-            var tests = new List<TestCase>();
+            var testColletion = new TestCollection();
 
             foreach (var testGroup in sources.GroupBy(x => sourceToProjSettings[x])) {
-                PythonProjectSettings settings = testGroup.Key;
-                if(!settings.PytestEnabled) {
-                    continue;
-                }
-
-                var discovery = new DiscoveryService(frameworkHandle);
-                var results = discovery.RunDiscovery(settings, testGroup);
+                var settings = testGroup.Key;
 
-                if (_cancelRequested.WaitOne(0)) {
-                    return;
+                try {
+                    var discovery = DiscovererFactory.GetDiscoverer(settings);
+                    discovery.DiscoverTests(testGroup, frameworkHandle, testColletion);
```

Comment:
new TestCollection() class is used here in the Executor when vstest is run from commandline.  TestCollection is just an implementation of ITestCaseDiscoverySink and just holds a list of testcases